### PR TITLE
Histogram: Respect min/max panel settings for x-axis

### LINKED ISSUE #62237: Histogram Visualization (min and max) option does not work
**What happened**:

This issue is opened after multiple community users reported it [[1]](https://community.grafana.com/t/does-min-and-max-standard-options-work-with-histograms/75380) and [[2]](https://community.grafana.com/t/histogram-charts-error-with-x-axis-min-max-values/23549).

The Histogram Visualization does not seem to pick the values from the min and max option

**What you expected to happen**:

While using the Histogram Visual panel, and setting an option for Min and Max from inside the Standard Option seems to have no effect.

**How to reproduce it (as minimally and precisely as possible)**:

image
image

**Anything else we need to know?**:

Not 100% sure if this is a known issue, but the closest match I found is this similar issue:(https://github.com/grafana/grafana/discussions/58503#discussioncomment-4737854)

**Environment**:
- Grafana version: 8.5.10 and 9.2.3 
- Data source type & version: Any
- OS Grafana is installed on: Linux OS
- User OS & Browser: Any

### PR DESCRIPTION
**Pull request by leeoniya**
Fixes https://github.com/grafana/grafana/issues/62237

<details><summary>histogram-min-max.json</summary>

```json
{
  "annotations": {
    "list": [
      {
        "builtIn": 1,
        "datasource": {
          "type": "grafana",
          "uid": "-- Grafana --"
        },
        "enable": true,
        "hide": true,
        "iconColor": "rgba(0, 211, 255, 1)",
        "name": "Annotations & Alerts",
        "type": "dashboard"
      }
    ]
  },
  "editable": true,
  "fiscalYearStartMonth": 0,
  "graphTooltip": 0,
  "id": 672,
  "links": [],
  "liveNow": false,
  "panels": [
    {
      "datasource": {
        "type": "testdata",
        "uid": "PD8C576611E62080A"
      },
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "palette-classic"
          },
          "custom": {
            "axisCenteredZero": false,
            "axisColorMode": "text",
            "axisLabel": "",
            "axisPlacement": "auto",
            "barAlignment": 0,
            "drawStyle": "line",
            "fillOpacity": 0,
            "gradientMode": "none",
            "hideFrom": {
              "legend": false,
              "tooltip": false,
              "viz": false
            },
            "lineInterpolation": "linear",
            "lineWidth": 1,
            "pointSize": 5,
            "scaleDistribution": {
              "type": "linear"
            },
            "showPoints": "auto",
            "spanNulls": false,
            "stacking": {
              "group": "A",
              "mode": "none"
            },
            "thresholdsStyle": {
              "mode": "off"
            }
          },
          "mappings": [],
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              },
              {
                "color": "red",
                "value": 80
              }
            ]
          }
        },
        "overrides": []
      },
      "gridPos": {
        "h": 15,
        "w": 12,
        "x": 0,
        "y": 0
      },
      "id": 1,
      "options": {
        "legend": {
          "calcs": [],
          "displayMode": "list",
          "placement": "bottom",
          "showLegend": true
        },
        "tooltip": {
          "mode": "single",
          "sort": "none"
        }
      },
      "title": "Panel Title",
      "type": "timeseries"
    },
    {
      "datasource": {
        "type": "datasource",
        "uid": "-- Dashboard --"
      },
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "thresholds"
          },
          "custom": {
            "fillOpacity": 50,
            "gradientMode": "none",
            "hideFrom": {
              "legend": false,
              "tooltip": false,
              "viz": false
            },
            "lineWidth": 1
          },
          "mappings": [],
          "max": 120,
          "min": 0,
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              }
            ]
          }
        },
        "overrides": []
      },
      "gridPos": {
        "h": 15,
        "w": 12,
        "x": 12,
        "y": 0
      },
      "id": 2,
      "options": {
        "bucketOffset": 0,
        "bucketSize": 2,
        "legend": {
          "calcs": [],
          "displayMode": "list",
          "placement": "bottom",
          "showLegend": true
        }
      },
      "targets": [
        {
          "datasource": {
            "type": "datasource",
            "uid": "-- Dashboard --"
          },
          "panelId": 1,
          "refId": "A"
        }
      ],
      "title": "Panel Title",
      "type": "histogram"
    }
  ],
  "refresh": "",
  "schemaVersion": 38,
  "style": "dark",
  "tags": [],
  "templating": {
    "list": []
  },
  "time": {
    "from": "now-6h",
    "to": "now"
  },
  "timepicker": {},
  "timezone": "",
  "title": "histogram-min-max",
  "uid": "bacbfa77-99ef-4bca-a029-4a0b60107f52",
  "version": 4,
  "weekStart": ""
}
```
</details>

https://github.com/grafana/grafana/assets/43234/0f210872-8bfe-4a53-bbd1-8fe48711ec68

### CODE REVIEW

**Comment by justinsmalley** [comment]
I hope this is not forgotten about. This is an important change for myself and many others. I am still using old graph panel histograms due to this missing functionality, however it looks like that will be deprecated soon. https://github.com/grafana/grafana/issues/63732

**Comment by SEngelnkemper** [comment]
Why are actually ignoring this obvious bug?

**Comment by kustrun** [comment]
This fix was not released yet?

**Comment by leeoniya** [comment]
for everyone here, can you tell me if/how you expect this to work any differently than simply pre-filtering your data using https://grafana.com/docs/grafana/latest/panels-visualizations/query-transform-data/transform-data/#filter-data-by-value

**Comment by leeoniya** [comment]
> e.g. what do you expect legend calcs to read when you apply these limits in the panel? should they be full range or filtered range?

actually i should be more specific here. the legend calcs always happen before the data hits the panel. so they will always be full range. would you find it confusing that your min/max/avg in the legend do not correspond to the min/max limits you set in the panel?

**Comment by justinsmalley** [comment]
My use case is that I want to set the max and min to levels outside the normal operating range.

As an example, imagine the output of solar panel. In full light, it can output 200W. On a cloudy day, it may only output a max of 100W. If I was to plot these a low light day and a full light day on a histogram without the same x-axis max value, the histogram would look roughly similar each day. But in reality, the behaviours are very different. Once the defined max and min x axis values are available, you could know if its a poor light day if the peak is further to the left, and a good light day if the peak is further to the right. There could also be two peaks if it suddenly turned very cloudy halfway through the day. 

image

In summary, setting a max and min on the x-axis allows you to get a feeling for not only the distribution, but also the mean and standard deviation.  

Regarding the legend, I haven't put any thought into that, especially with respect to applying transformations.

**Comment by leeoniya** [comment]
thanks. yeah, that makes sense. what's being asked for in https://community.grafana.com/t/does-min-and-max-standard-options-work-with-histograms/75380/3 is a combo of this. (grow the max and clip the min).

it sounds like if you want the legend to reflect any limits smaller than data min/max, then you should use transformations, and this PR should land as-is. so for the case linked above it would be transform/filter out values < 20, and then set panel to min: 20, max: 70.

**Review by ryantxu — APPROVED** [review]
